### PR TITLE
[PBI-1454] Fix error reporting issue and cleanup

### PR DESCRIPTION
**Pull request by dchauOoyala**
- Add function, _hasLinearAd() to check if the ad has a linear ad before calling _handleLinearAd()
- Add function, _hasNonLinearAd() to check if the ad has a linear ad before calling _handleNonLinearAd()
- Code Cleanup and JSDoc fix

### CODE REVIEW

File: js/ad_manager_vast.js [hunk 1]
```diff
@@ -1067,6 +1067,40 @@ OO.Ads.manager(function(_, $) {
...
 
     /**
+     * Helper function to determine if the ad is a linear ad.
+     * @private
+     * @method Vast#_hasLinearAd
+     * @param {object} ad The ad object
+     * @returns {boolean} true if the ad is a linear ad, false otherwise.
+     */
+    var _hasLinearAd = _.bind(function(ad) {
+      var linearObjectKeys = _.keys(ad.linear);
```
Comment (gfrankOoyala) [inline js/ad_manager_vast.js]: what if there is no clickthrough? Should we check the creative instead?

File: js/ad_manager_vast.js [hunk 2]
```diff
@@ -1067,6 +1067,40 @@ OO.Ads.manager(function(_, $) {
...
+     * @returns {boolean} true if the ad is a nonlinear ad, false otherwise.
+     */
+    var _hasNonLinearAd = _.bind(function(ad) {
+      var nonLinearObjectKeys = _.keys(ad.nonLinear);
+      // Magic number 1 because after mergeVastAdResult() is called, the ad object
+      // has will have an object associated to the key, "nonLinear". This object always
+      // has at least 1 key (tracking) even if a nonlinear creative does not exist. If one
+      // does exist, there must be more than 1 key; there must be a "url" key as specified
+      // by the Vast 3.0 specifications.
+      return (nonLinearObjectKeys.length > 1);
```
Comment (gfrankOoyala) [inline js/ad_manager_vast.js]: What if there is a clickthrough?

**Comment by gfrankOoyala** [comment]
Why not use the isLinear functions?

File: js/ad_manager_vast.js (hunk 1 above)
Comment (dchauOoyala) [inline js/ad_manager_vast.js]: so the way it originally was in our code, clickTracking and tracking will always be available in the linear object. It's not great, so I'm in the middle of changing it so I don't need to use these magic values of 1/2

**Comment by dchauOoyala** [comment]
The isLinear properties are only available when the ad is played. I do not think the isLinear property is available to whatever "ad" object is in context when we call handleLinearAd/handleNonLinearAd

**Comment by gfrankOoyala** [comment]
Sure but instead of checking clickthrough and tracking why not check for creative instead? Seems like assuming 2 or 1 is a magic number is not accurate. What if there is no clickthrough or error tracking urls but a creative none the less?

**Comment by dchauOoyala** [comment]
I see what you're saying, that's actually the way I originally made hasLinearAd/hasNonLinearAd work; check the vastXML to see if the creative exists. But, that got lost in the midst of merging with the podded ads code. I can do it that way instead of checking against the ad object.

**Comment by aeng7** [comment]
LGTM

**Pull request merged**